schema/gen/go/genStructReprStringjoin.go

```diff
@@ -0,0 +1,246 @@
+	AdjCfg *AdjunctCfg
+	mixins.StringAssemblerTraits
+	PkgName string
+	Type    schema.TypeStruct
+}
+
+func (g structReprStringjoinReprBuilderGenerator) EmitNodeBuilderType(w io.Writer) {
+	doTemplate(`
+		type _{{ .Type | TypeSymbol }}__ReprBuilder struct {
+			_{{ .Type | TypeSymbol }}__ReprAssembler
+		}
+	`, w, g.AdjCfg, g)
+}
+func (g structReprStringjoinReprBuilderGenerator) EmitNodeBuilderMethods(w io.Writer) {
+	doTemplate(`
+		func (nb *_{{ .Type | TypeSymbol }}__ReprBuilder) Build() ipld.Node {
+			return nb.w
+		}
+		func (nb *_{{ .Type | TypeSymbol }}__ReprBuilder) Reset() {
+			var w _{{ .Type | TypeSymbol }}
+			*nb = _{{ .Type | TypeSymbol }}__ReprBuilder{_{{ .Type | TypeSymbol }}__ReprAssembler{w: &w, fcb:nb.fcb_root}}
+		}
+		func (nb *_{{ .Type | TypeSymbol }}__ReprBuilder) fcb_root() error {
+			if nb.z == true {
+				return mixins.StringAssembler{"{{ .PkgName }}.{{ .TypeName }}.Repr"}.AssignNull()
+			}
+			return nil
+		}
+	`, w, g.AdjCfg, g)
+	// Generate a single-step construction function -- this is easy to do for a scalar,
+	//  and all representations of scalar kind can be expected to have a method like this.
+	// The function is attached to the nodestyle for convenient namespacing;
+	//  it needs no new memory, so it would be inappropriate to attach to the builder or assembler.
+	// The function is directly used internally by anything else that might involve recursive destructuring on the same scalar kind
+	//  (for example, structs using stringjoin strategies that have one of this type as a field, etc).
+	// Since we're a representation of scalar kind, and can recurse,
+	//  we ourselves presume this plain construction method must also exist for all our members.
+	// REVIEW: We could make an immut-safe verion of this and export it on the NodeStyle too, as `FromString(string)`.
+	// FUTURE: should engage validation flow.
+	doTemplate(`
+		func (_{{ .Type | TypeSymbol }}__ReprStyle) fromString(w *_{{ .Type | TypeSymbol }}, v string) error {
+			ss, err := mixins.SplitExact(v, "{{ .Type.RepresentationStrategy.GetDelim }}", {{ len .Type.Fields }})
+			if err != nil {
+				return ipld.ErrUnmatchable{TypeName:"{{ .PkgName }}.{{ .Type.Name }}.Repr", Reason: err}
+			}
+			{{- $dot := . -}} {{- /* ranging modifies dot, unhelpfully */ -}}
+			{{- range $i, $field := .Type.Fields }}
+			if err := (_{{ $field.Type | TypeSymbol }}__ReprStyle{}).fromString(&w.{{ $field | FieldSymbolLower }}, ss[{{ $i }}]); err != nil {
+				return ipld.ErrUnmatchable{TypeName:"{{ $dot.PkgName }}.{{ $dot.Type.Name }}.Repr", Reason: err}
+			}
+			{{- end}}
+			return nil
+		}
+	`, w, g.AdjCfg, g)
+}
+func (g structReprStringjoinReprBuilderGenerator) EmitNodeAssemblerType(w io.Writer) {
+	// - 'w' is the "**w**ip" pointer.
+	// - 'z' is used to denote a null (in case we're used in a context that's acceptable).  z for **z**ilch.
+	// - 'fcb' is the **f**inish **c**all**b**ack, supplied by the parent if we're a child assembler.
```

Comment:
FIXME(self): this is both out of date and clearly not covered by tests :/

Current tests for structs with stringjoin representation cover type-level assemblers but apparently not representation-level assemblers